File: lista2/sorts/rand_gen.cpp
#include <iostream>
#include <vector>
#include <random>

using namespace std;

vector<int> generateRandomKeys(int n) {
    vector<int> keys;
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> dis(0, 2 * n - 1);

    for (int i = 0; i < n; ++i) {
        keys.push_back(dis(gen));
    }

    return keys;
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        cerr << "Usage: " << argv[0] << " <array_size>" << endl;
        return 1;
    }

    int n = stoi(argv[1]);
    if (n <= 0) {
        cerr << "Array size must be a positive integer." << endl;
        return 1;
    }

    vector<int> randomKeys = generateRandomKeys(n);
    for (int key : randomKeys) {
        cout << key << " ";
    }
    cout << endl;

    return 0;
}

File: lista3/zad5/DualPivotQuickSort/RandGen.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class RandGen {

    public static List<Integer> generateRandomKeys(int n) {
        List<Integer> keys = new ArrayList<>();
        Random rand = new Random();

        for (int i = 0; i < n; ++i) {
            keys.add(rand.nextInt(2 * n));
        }

        return keys;
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Usage: java RandGen <array_size>");
            System.exit(1);
        }

        int n = Integer.parseInt(args[0]);
        if (n <= 0) {
            System.err.println("Array size must be a positive integer.");
            System.exit(1);
        }

        List<Integer> randomKeys = generateRandomKeys(n);
        for (int key : randomKeys) {
            System.out.print(key + " ");
        }
        System.out.println();
    }
}
